catalog/schematron/catalog-schematron-plugin/src/main/java/ddf/services/schematron/SchematronValidationService.java

```diff
@@ -74,537 +71,162 @@
-                        trimmedErrors.add(error.trim());
-                    }
-                    // If warnings are to be included from the Schematron report
-                    // as part of the errors
-                    // message
-                    List<String> trimmedWarnings = new ArrayList<>();
-                    if (!this.suppressWarnings) {
-                        warnings = this.report.getWarnings();
-                        LOGGER.debug("warnings.size() = {}", warnings.size());
-                        for (String warning : warnings) {
-                            LOGGER.debug("warning = {}", warning);
-                            errorMessage.append(warning);
-                            errorMessage.append("\n");
-                            trimmedWarnings.add(warning.trim());
-                        }
-                    }
-                    LOGGER.debug(errorMessage.toString());
-                    throw new SchematronValidationException("Schematron validation failed.",
-                            trimmedErrors, trimmedWarnings);
-                }
-
-            } catch (TransformerException te) {
-                LOGGER.warn("Could not setup validator to perform validation", te);
+        for (Templates validator : validators) {
+            String metadata = metacard.getMetadata();
+            if (metadata == null) {
                 throw new SchematronValidationException(
-                        "Could not setup validator to perform validation.");
+                        "The Metacard.METADATA attribute must not be null to run schematron validation against the Metacard");
+            }
+            if (namespace != null && !namespace.equals(XMLUtils.getRootNamespace(metadata))) {
+                return;
             }
-        } else {
-            throw new SchematronValidationException(
-                    "The Metacard.METADATA attribute must not be null to run shematron validation against the Metacard");
+            schematronReport = generateReport(metadata, validator);
+            if (!schematronReport.isValid(suppressWarnings)) {
+                throw new SchematronValidationException("Schematron validation failed.", schematronReport.getErrors(), schematronReport.getWarnings());
+            }
+        }
+    }
+
+    private SchematronReport generateReport(String metadata, Templates validator) throws SchematronValidationException {
+        StringReader metadataReader = new StringReader(metadata);
+
+        SchematronReport report;
+        try {
+            Transformer transformer = validator.newTransformer();
+            DOMResult schematronResult = new DOMResult();
+            transformer.transform(new StreamSource(metadataReader), schematronResult);
+            report = new SvrlReport(schematronResult);
+        } catch (TransformerException e) {
+            throw new SchematronValidationException("Could not setup validator to perform validation.");
```

Comment:
It's fine to re-throw more descriptively, but I think the original exception should still be included in the re-throw.